maltoolbox/attackgraph/attackgraph.py

```diff
@@ -41,11 +41,11 @@ def _process_step_expression(lang_graph: LanguageGraph, model: Model,
     A tuple pair containing a list of all of the target assets and the name of
     the attack step.
     """
-    logger.debug('Processing Step Expression:\n' \
-        + json.dumps(step_expression, indent = 2))
 
-
-    lang = lang_graph._lang_spec
+    if logger.level == logging.DEBUG:
+        # This expressions takes a lot of time
+        logger.debug('Processing Step Expression:\n' \
+            + json.dumps(step_expression, indent = 2))
```

Comment:
The [suggested way](https://docs.python.org/3/howto/logging.html#optimization) to do this is:

```
if logger.isEnabledFor(logging.DEBUG):
    ...
```

Also, the suggested way to format strings in log statements is with `%s` and passing the variables as arguments, like `debug("string %s %s", arg1, arg2)`, e.g.:

```
     logger.debug('Processing Step Expression:\n%s',
            json.dumps(step_expression, indent = 2))
```

So that the string gets formatted only if the debug level is enabled. The dumps call, however, will still be evaluated regardless of the debug level and this is why you need to conditional check anyway.

But adding the conditional check all the time (given that there are a lot of debug statements in the code) seems cumbersome to me. I would create a custom logging formatter, use the suggested way to format log strings and dump to json in the formatter's `format()` method only if an argument is a dict. Or, I would send a keyword arg along side the positional args and check that for dumping to json, e.g.:

```
debug("string %s %s", str_var1, str_var2, json=True)
```

Reply:
Thanks, I will change the if statement.
Yes, I think all of our logging statements to use the %s-method instead, but I think it should be a future PR for that. Same with your suggestion on a custom formatter. Do you agree we can do it at a later time?

Reply:
Created PR for using format strings in logging, working on it #48